Source/Modules/python.cxx

```diff
@@ -3307,18 +3312,19 @@ class PYTHON:public Language {
       Wrapper_print(f, f_wrappers);
     }
 
+    bool use_static_method = flat_static_method || !in_class || constructor || (Cmp(storage, "friend") == 0);
```

Comment:
The variable name is confusing and subsequent logic is hard to follow to me. You really ought to be using a flag based on whether the method is a static method, so something like:
```
bool generate_flat_static_method = flat_static_method && Swig_storage_isstatic(n);
```
and then adjust the subsequent logic on this.

Swig_storage_isstatic(n) is simply checking the "storage" attribute contains "static" and is always set for static methods.